### PR TITLE
[SPARK-12746][ML] ArrayType(_, true) should also accept ArrayType(_, false) fix for branch-1.6

### PR DESCRIPTION
**Pull request by Earthson**
https://issues.apache.org/jira/browse/SPARK-13359

### CODE REVIEW

**Comment by Earthson** [comment]
@mengxr Hi, Xiangrui, I've copied the code from master to fix SPARK-12746 for branch-1.6.

related: https://github.com/apache/spark/pull/10697

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test.

**Comment by srowen** [comment]
Please tag this for SPARK-12746 instead

**Comment by Earthson** [comment]
https://issues.apache.org/jira/browse/SPARK-12746

File: mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala [hunk 1]
```diff
@@ -44,6 +44,23 @@ private[spark] object SchemaUtils {
...
   /**
+    * Check whether the given schema contains a column of one of the require data types.
+    * @param colName  column name
+    * @param dataTypes  required column data types
+    */
+  def checkColumnTypes(
+                        schema: StructType,
+                        colName: String,
+                        dataTypes: Seq[DataType],
+                        msg: String = ""): Unit = {
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala]: indent

**Comment by Earthson** [comment]
Jenkins, test this please.

**Comment by Earthson** [comment]
@srowen Is this ok to merge?

**Comment by srowen** [comment]
OK by me though ideally @mengxr would merge. I'll do it tomorrow if not

**Comment by mengxr** [comment]
Merged into branch-1.6. Thanks!

**Pull request closed without merging**